model_compression_toolkit/core/pytorch/hessian/weights_trace_hessian_calculator_pytorch.py

```diff
@@ -14,37 +14,109 @@
 # ==============================================================================
 
 from typing import List
-
-
+import torch
+from torch import autograd
+import numpy as np
 from model_compression_toolkit.core.common import Graph
-from model_compression_toolkit.core.common.hessian import TraceHessianRequest
+from model_compression_toolkit.core.common.hessian import TraceHessianRequest, HessianInfoGranularity
 from model_compression_toolkit.core.pytorch.hessian.trace_hessian_calculator_pytorch import \
     TraceHessianCalculatorPytorch
 from model_compression_toolkit.logger import Logger
-import torch
+from model_compression_toolkit.core.pytorch.back2framework.float_model_builder import FloatPyTorchModelBuilder
+from model_compression_toolkit.core.pytorch.default_framework_info import DEFAULT_PYTORCH_INFO
+from model_compression_toolkit.constants import HESSIAN_NUM_ITERATIONS, MIN_JACOBIANS_ITER, JACOBIANS_COMP_TOLERANCE, HESSIAN_EPS
+
 
 class WeightsTraceHessianCalculatorPytorch(TraceHessianCalculatorPytorch):
     """
-    Pytorch-specific implementation of the Trace Hessian approximation computation w.r.t to a node's weights.
+    Pytorch-specific implementation of the Trace Hessian approximation computation w.r.t node's weights.
     """
 
     def __init__(self,
                  graph: Graph,
                  input_images: List[torch.Tensor],
                  fw_impl,
-                 trace_hessian_request: TraceHessianRequest):
+                 trace_hessian_request: TraceHessianRequest,
+                 num_iterations_for_approximation: int = HESSIAN_NUM_ITERATIONS):
         """
 
         Args:
             graph: Computational graph for the float model.
             input_images: List of input images for the computation.
             fw_impl: Framework-specific implementation for trace Hessian computation.
             trace_hessian_request: Configuration request for which to compute the trace Hessian approximation.
+            num_iterations_for_approximation: Number of iterations to use when approximating the Hessian trace.
         """
         super(WeightsTraceHessianCalculatorPytorch, self).__init__(graph=graph,
                                                                    input_images=input_images,
                                                                    fw_impl=fw_impl,
-                                                                   trace_hessian_request=trace_hessian_request)
+                                                                   trace_hessian_request=trace_hessian_request,
+                                                                   num_iterations_for_approximation=num_iterations_for_approximation)
+
+
+    def compute(self) -> np.ndarray:
+        """
+        Compute the Hessian-based scores w.r.t target node's weights.
```

Comment:
Please add to the docs some examples of returned shapes based on the requested granularity.